File: Problem004/go/main.go
package main

import (
	"fmt"
	"strconv"
)

func isPalindrome(n int) bool {
	num := strconv.Itoa(n)
	for i := 0; i < len(num)/2; i++ {
		if num[i] != num[len(num)-i-1] {
			return false
		}
	}
	return true
}

func main() {

	max := 0
	for i := 100; i < 1000; i++ {
		for j := 100; j < 1000; j++ {
			if isPalindrome(i * j) {
				if max < i*j {
					max = i * j
				}
			}
		}
	}
	fmt.Println(max)
}

File: Problem004/java/Main.java
class Main{
	public static void main(String[] args){
		int max = 0;
		for(int i=100;i<=999;i++){
			for(int j=100;j<=999;j++){
				max = (isKaibun(i*j) && i*j > max) ? i*j : max;
			}
		}
		System.out.println(max);
	}
	
	public static boolean isKaibun(int n){
		String str = n+"";
		for(int i=0;i<str.length()/2;i++){
			if(str.charAt(i) != str.charAt(str.length() - i - 1))return false;
		}
		return true;
	}
}